Assets/MixedRealityToolkit/Inspectors/MixedRealityShaderGUI.cs

```diff
@@ -0,0 +1,274 @@
+                    }
+                    break;
+
+                case RenderingMode.Custom:
+                    {
+                        material.SetOverrideTag(BaseStyles.renderTypeName, BaseStyles.customRenderingModeNames[(int)customMode]);
+                        // _SrcBlend, _DstBlend, _BlendOp, _ZTest, _ZWrite, _ColorWriteMask are controlled by UI.
+
+                        switch (customMode)
+                        {
+                            case CustomRenderingMode.Opaque:
+                                {
+                                    material.DisableKeyword(BaseStyles.alphaTestOnName);
+                                    material.DisableKeyword(BaseStyles.alphaBlendOnName);
+                                }
+                                break;
+
+                            case CustomRenderingMode.TransparentCutout:
+                                {
+                                    material.EnableKeyword(BaseStyles.alphaTestOnName);
+                                    material.DisableKeyword(BaseStyles.alphaBlendOnName);
+                                }
+                                break;
+
+                            case CustomRenderingMode.Transparent:
+                                {
+                                    material.DisableKeyword(BaseStyles.alphaTestOnName);
+                                    material.EnableKeyword(BaseStyles.alphaBlendOnName);
+                                }
+                                break;
+                        }
+
+                        material.renderQueue = (renderQueueOverride >= 0) ? renderQueueOverride : material.renderQueue;
+                    }
+                    break;
+            }
+        }
+
+        protected static bool PropertyEnabled(MaterialProperty property)
+        {
+            return !property.floatValue.Equals(0.0f);
+        }
+
+        protected static float? GetFloatProperty(Material material, string propertyName)
+        {
+            if (material.HasProperty(propertyName))
+            {
+                return material.GetFloat(propertyName);
+            }
+
+            return null;
+        }
+
+        protected static Vector4? GetVectorProperty(Material material, string propertyName)
+        {
+            if (material.HasProperty(propertyName))
+            {
+                return material.GetVector(propertyName);
+            }
+
+            return null;
+        }
+
+        protected static Color? GetColorProperty(Material material, string propertyName)
+        {
+            if (material.HasProperty(propertyName))
+            {
+                return material.GetColor(propertyName);
+            }
+
+            return null;
+        }
+
+        protected static void SetFloatProperty(Material material, string keywordName, string propertyName, float? propertyValue)
```

Comment:
>SetFloatProperty [](start = 30, length = 16)

Nit: It might be worth adding a doccomment on this function, if only because it seems to be doing more than just setting a float property (i.e. also optionally enabling the keyword based on the property value itself)

Reply:
I see that this was taken from the existing standard shader... I guess it's up to you here :)

---
In reply to: [296323173](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4993#discussion_r296323173) [](ancestors = 296323173)